src/librustc/hir/print.rs

```diff
@@ -2341,14 +2331,12 @@ impl<'a> State<'a> {
                                 -> io::Result<()> {
         self.s.word(&visibility_qualified(vis, ""))?;
 
-        match header.constness {
-            hir::Constness::NotConst => {}
-            hir::Constness::Const => self.word_nbsp("const")?,
+        if let hir::Constness::Const = header.constness {
+            self.word_nbsp("const")?
         }
 
-        match header.asyncness {
-            hir::IsAsync::NotAsync => {}
-            hir::IsAsync::Async => self.word_nbsp("async")?,
+        if let hir::IsAsync::Async = header.asyncness {
+            self.word_nbsp("async")?
```

Comment:
The same here and `constness` above. It's a bit more verbose to list all variants, but also a lot more readable for someone who does not know the code base by heart.